lib/redix.ex

```diff
@@ -791,7 +791,20 @@ defmodule Redix do
   end
 
   defp pipeline_without_checks(conn, commands, opts) do
-    Redix.Connection.pipeline(conn, commands, opts[:timeout] || @default_timeout)
+    timeout = opts[:timeout] || @default_timeout
+    {elapsed_time, result} = :timer.tc(Redix.Connection, :pipeline, [conn, commands, timeout])
```

Comment:
Having `start_time = System.system_time()` in the metadata might be nice, it's not really possible to derive an accurate start time in the handler from just the duration, since it'd include any processing that happens in the various telemetry handlers...

It wouldn't be used to generate metrics, but it could be really useful for telemetry handlers that are doing tracing

Reply:
Great ideas. Implemented them!